proposals/0019-data-availability.md

```diff
@@ -0,0 +1,32 @@
+---
+simd: '0019'
+title: Data Availability
+authors:
+  - Eclipse Laboratories
+category: Standard
+type: Core
+status: Draft
+created: 2022-01-10
+---
+
+## Summary
+We propose a new transaction type with a higher size limit that is not intended for execution.
+
+## Motivation
+Multiple rollups are looking to use Solana as a data availability layer to post rollup blocks to, including Eclipse Labs and Layer N. This is a poorly supported use-case—the only solution currently requires writing chunks of the data one-by-one to an on-chain account. In addition, since it requires minimal runtime execution/processing by the full node, the price of posting data would ideally be more in line with the cost of rent for account storage. This feature would allow rollups to seamlessly post blocks of ordered transactions to be finalized by Solana as an L1.
+
+## Alternatives Considered
+A new transaction type is not strictly necessary for Solana to support data availability. However, due to the hard limit on transaction size, a .1 MB block, for example, would require >80 transactions to upload on-chain. In addition, rollup data does not need to be written to state in the execution environment itself, but only a commitment to the data.
```

Comment:
I am not sure there is a difference. By commitment to the state do you mean the ledger only?

Reply:
Yes that's what we meant